listenbrainz/webserver/views/api_tools.py

```diff
@@ -58,49 +60,46 @@ def _send_listens_to_queue(listen_type, listens):
+                current_app.logger.error("Redis rpush playing_now write error: " + str(e))
                 raise InternalServerError("Cannot record playing_now at this time.")
         else:
             submit.append(listen)
 
     if submit:
-        while True:
-            try:
-                channel = rabbitmq_connection._rabbitmq.channel()
-                channel.exchange_declare(exchange='incoming', type='fanout')
-                channel.queue_declare('incoming', durable=True)
-            except pika.exceptions.ConnectionClosed as e:
-                # RabbitMQ connection closed, so re-establish and try again
-                current_app.logger.error("Connection to rabbitmq closed while creating channel: %s" % str(e))
-                init_rabbitmq_connection(current_app)
-                continue
-            except pika.exceptions.NoFreeChannels as e:
-                if channel:
-                    channel.close() # just in case
-                current_app.logger.error("Cannot create a rabbitmq channel: %s" % str(e))
-                raise ServiceUnavailable("Cannot submit listens to queue, please try again later.")
-
-            # There is no good documentation on what exceptions could be raised here. Loads apparently,
-            # this catching blanco exceptions
-            try:
-                channel.basic_publish(exchange='incoming', routing_key='', body=ujson.dumps(submit),
-                    properties=pika.BasicProperties(delivery_mode = 2, ))
-            except pika.exceptions.ConnectionClosed as e:
-                # RabbitMQ connection closed, so re-establish and try again
-                current_app.logger.error("Connection to rabbitmq closed while trying to publish: %s" % str(e))
-                init_rabbitmq_connection(current_app)
-                continue
-            except Exception as e:
-                channel.close()
-                current_app.logger.error("Cannot publish to rabbitmq channel: %s" % str(e))
-                raise ServiceUnavailable("Cannot submit listens to queue, please try again later.")
-
-            _redis.redis.incr(INCOMING_QUEUE_SIZE_KEY, len(submit))
+        try:
+            channel = rabbitmq_connection._rabbitmq.channel()
+            channel.exchange_declare(exchange='incoming', type='fanout')
+            channel.queue_declare('incoming', durable=True)
+        except pika.exceptions.ConnectionClosed as e:
+            current_app.logger.error("Connection to rabbitmq closed while creating channel: %s" % str(e))
+            raise ServiceUnavailable("Cannot submit listens to queue, please try again later.")
+        except pika.exceptions.NoFreeChannels as e:
+            if channel:
+                channel.close() # just in case
+            current_app.logger.error("Cannot create a rabbitmq channel: %s" % str(e))
+            raise ServiceUnavailable("Cannot submit listens to queue, please try again later.")
```

Comment:
Nope, HTML is what is returned as of now because of these error handlers: https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/webserver/errors.py#L53